src/activities/Elsa.Activities.Mqtt/Services/MqttClientWrapper.cs

```diff
@@ -25,25 +25,22 @@ public MqttClientWrapper(IMqttClient client, Options.MqttClientOptions options,
             _logger = logger;
         }
 
-        static bool subscribed = false;
-
-        public async Task SubscribeAsync(string topic)
+        
+        private async Task SubscribeAsync(string topic, Func<MqttApplicationMessage, Task> handler)
         {
             if (!Client.IsConnected)
             {
                 var opt = Options.GenerateMqttClientOptions();
                 Client.ConnectAsync(opt).Wait(); //Sync
-
+                _messageHandler = handler;
+                await Client.SubscribeAsync(topic, Options.QualityOfService);
```

Comment:
Isn't it safer to subscribe to the topic **after** setting up the event handler, instead of the other way around?
In theory, it is possible that the client immediately receives an event after subscribing, at which point the next line where the event handler is assigned, hasn't executed yet. Admittedly, this seems like something that wouldn't actually happen in practice, but if it doesn't hurt, maybe better to first attach the handler, then subscribe to the topic.

Reply:
you are right , im scared to leave handler binded but need to se handler before